File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int> mymap;

        for(auto x:nums){
            if(mymap.find(x)!=mymap.end()){
                return true;
            }
            mymap[x]++;
        }

        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> myMap = new HashMap<>();

        for(int num:nums){
            myMap.put(num, myMap.getOrDefault(num,0)+1);
        }

        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b)->(myMap.get(a)-myMap.get(b)));

        for(int num:myMap.keySet()){
            pq.offer(num);
            if(pq.size()>k){
                pq.poll();
            }
        }
        int[] result = new int[k];
        for(int i=k-1;i>=0;i--){
            result[i]=pq.poll();
        }
        return result;
    }
}
